thingif/src/main/java/com/kii/thingif/internal/gson/ActionAdapter.java

```diff
@@ -65,9 +66,21 @@ public Action deserialize(JsonElement jsonElement, Type typeOfT, JsonDeserializa
 
         Map.Entry<String, JsonElement> firstEntry = jsonElement.getAsJsonObject().entrySet().iterator().next();
         JsonElement actionValue = firstEntry.getValue();
-        Field firstField = this.actionClass.getDeclaredFields()[0];
-        String fieldName = firstField.getName();
+
+        String fieldName = null;
+
+        if (this.actionClass.getEnclosingClass() != null &&
+                !Modifier.isStatic(this.actionClass.getModifiers())) {
+            //never happen, since action class was validated when registered.
+            throw new JsonParseException("non static inner class is not allowed");
+        } else if (this.actionClass.getDeclaredFields().length == 1){
+            fieldName = this.actionClass.getDeclaredFields()[0].getName();
+        }
```

Comment:
I prefer to check `actionClass` in constructor because if `ActionAdapter#deserialize()` is called twice, then checking action process runs twice. It is redundant.

If you prefer to check this here, I follow your decision:+1: